asv_bench/benchmarks/eval.py

```diff
@@ -3,192 +3,36 @@
-        pd.eval('df + df2 + df3 + df4', engine='python')
-
-
-class eval_frame_and_all_threads(object):
-    goal_time = 0.2
-
-    def setup(self):
-        self.df = DataFrame(np.random.randn(20000, 100))
-        self.df2 = DataFrame(np.random.randn(20000, 100))
-        self.df3 = DataFrame(np.random.randn(20000, 100))
-        self.df4 = DataFrame(np.random.randn(20000, 100))
-
-    def time_eval_frame_and_all_threads(self):
-        pd.eval('(df > 0) & (df2 > 0) & (df3 > 0) & (df4 > 0)')
-
-
-class eval_frame_and_python_one_thread(object):
-    goal_time = 0.2
-
-    def setup(self):
-        self.df = DataFrame(np.random.randn(20000, 100))
-        self.df2 = DataFrame(np.random.randn(20000, 100))
-        self.df3 = DataFrame(np.random.randn(20000, 100))
-        self.df4 = DataFrame(np.random.randn(20000, 100))
-        expr.set_numexpr_threads(1)
-
-    def time_eval_frame_and_python_one_thread(self):
-        pd.eval('(df > 0) & (df2 > 0) & (df3 > 0) & (df4 > 0)', engine='python')
-
-
-class eval_frame_and_python(object):
-    goal_time = 0.2
-
-    def setup(self):
-        self.df = DataFrame(np.random.randn(20000, 100))
-        self.df2 = DataFrame(np.random.randn(20000, 100))
-        self.df3 = DataFrame(np.random.randn(20000, 100))
-        self.df4 = DataFrame(np.random.randn(20000, 100))
-
-    def time_eval_frame_and_python(self):
-        pd.eval('(df > 0) & (df2 > 0) & (df3 > 0) & (df4 > 0)', engine='python')
-
-
-class eval_frame_chained_cmp_all_threads(object):
-    goal_time = 0.2
-
-    def setup(self):
-        self.df = DataFrame(np.random.randn(20000, 100))
-        self.df2 = DataFrame(np.random.randn(20000, 100))
-        self.df3 = DataFrame(np.random.randn(20000, 100))
-        self.df4 = DataFrame(np.random.randn(20000, 100))
-
-    def time_eval_frame_chained_cmp_all_threads(self):
-        pd.eval('df < df2 < df3 < df4')
-
-
-class eval_frame_chained_cmp_python_one_thread(object):
-    goal_time = 0.2
+    params = [['numexpr', 'python'], [1, 'all']]
+    param_names = ['engine', 'threads']
 
-    def setup(self):
-        self.df = DataFrame(np.random.randn(20000, 100))
-        self.df2 = DataFrame(np.random.randn(20000, 100))
-        self.df3 = DataFrame(np.random.randn(20000, 100))
-        self.df4 = DataFrame(np.random.randn(20000, 100))
-        expr.set_numexpr_threads(1)
-
-    def time_eval_frame_chained_cmp_python_one_thread(self):
-        pd.eval('df < df2 < df3 < df4', engine='python')
-
-
-class eval_frame_chained_cmp_python(object):
-    goal_time = 0.2
-
-    def setup(self):
+    def setup(self, engine, threads):
         self.df = DataFrame(np.random.randn(20000, 100))
         self.df2 = DataFrame(np.random.randn(20000, 100))
         self.df3 = DataFrame(np.random.randn(20000, 100))
         self.df4 = DataFrame(np.random.randn(20000, 100))
 
-    def time_eval_frame_chained_cmp_python(self):
-        pd.eval('df < df2 < df3 < df4', engine='python')
+        if threads == 1:
```

Comment:
hmm maybe should set this explicity each time something is run to 1 (then you can set in this particular one to higher)